### PR TITLE
Build java 21 from temurin archives for debian 12

### PR DESCRIPTION
**Pull request by loosebazooka**
This is "experimental" and we shouldn't advertise it.

TODO:
1. auto updater: check github releases for a new version and generate an appropriate java_archives.bzl
2. eventually cover java 11, 17 on debian 12 with temurin
3. ask someone with intimate knowledge of bazel how to get version from the custom repository_rule into the env of the container

### CODE REVIEW

**Comment by loosebazooka** [comment]
@mihaimaruseac could use your bazel eyes if you have a chance

File: java_archives.bzl [hunk 1]
```diff
@@ -0,0 +1,62 @@
+load("//private/remote:temurin_archive.bzl", "temurin_archive")
+
+def repositories():
+    temurin_archive(
+        name = "temurin21_jre_amd64",
+        sha256 = "277f4084bee875f127a978253cfbaad09c08df597feaf5ccc82d2206962279a3",
+        strip_prefix = "jdk-21.0.1+12-jre",
+        urls = ["https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1+12/OpenJDK21U-jre_x64_linux_hotspot_21.0.1_12.tar.gz"],
+        version = "21.0.1+12",
```
Comment (loosebazooka) [inline java_archives.bzl:9]: @mihaimaruseac specifically, do you know how to make this value (version) accessible in another build file somewhere?

**Comment by loosebazooka** [comment]
@alos fyi

File: java_archives.bzl (hunk 1 above)
Comment (loosebazooka) [reply java_archives.bzl:20]: this is instead being exported to java/BUILD via JAVA_RELEASE_VERISONS

**Comment by loosebazooka** [comment]
alright, I think we worked through all the problems. This is ready @bobcallaway. @mihaimaruseac if you would like to officially review it, that would be very helpful.

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**